### PR TITLE
bugfix:min line in binary diff

### LINKED ISSUE #4: Parse full patches using git diff --binary
I have a diff from git that looks like this: 

```
diff --git a/foo.png b/foo.png
new file mode 100644
index 0000000..f91f164
Binary files /dev/null and b/foo.png differ
```

However, this is failing it's header parsing because in `parse_git_header` (https://github.com/cscorley/whatthepatch/blob/master/whatthepatch/patch.py#L186) there is a call to `findall_regex` looking to match any line in the diff to `git_header_old_line`, which matches on lines that are prefixed by `---`. Since this diff doesn't have that pattern in it, it doesn't attempt to parse the rest of the diff header. 

While it's true that committing binary files to git is a bit of an anti-pattern, small images like this are a common thing in git repos. Git is also capable of producing binary diffs by adding the `--binary` flag, which creates something more like this:

```
diff --git a/foo.png b/foo.png
new file mode 100644
index 0000000000000000000000000000000000000000..bac462fbda4b111ad571e7d8d4326e2e5f0c9365
GIT binary patch
literal 89791
<snip: tons of encoded binary data>
literal 0
HcmV?d00001
```

Not sure what the details of the values at `literal` are, but it seems to the the byte length of the decoded/uncompressed binary blob, and that final trailing bit `HcmV?d00001` is a complete mystery. This may be documented somewhere, but it's not super obvious from a Google search. 

Anyhow, both of those format fail header detection. I'm currently bypassing this by detecting/converting these lines into normal multi-line git file headers... but that's a PIA. :)

### PR DESCRIPTION
**Pull request by babenek**
fixes #4 
Fixed wrong assertion in minimal line length of binary diff

### CODE REVIEW

**Comment by abbradar** [comment]
I feel that the second part of the assertion should be `(len(line) - 1) % 5) == 0` per [the spec](https://web.archive.org/web/20200726102316/http://www.gelato.unsw.edu.au/archives/git/0605/19975.html).

File: src/whatthepatch/patch.py [hunk 1]
```diff
@@ -968,7 +968,7 @@ def parse_git_binary_diff(text):
                 continue
         elif new_size > 0:
             if base85string.match(line):
-                assert len(line) >= 7 and ((len(line) - 2) % 5) != 0
+                assert len(line) >= 6 and ((len(line) - 2) % 5) != 0
```
Comment (babenek) [inline src/whatthepatch/patch.py:971]:
```suggestion
                assert len(line) >= 6 and ((len(line) - 1) % 5) == 0
```

File: src/whatthepatch/patch.py [hunk 2]
```diff
@@ -992,7 +992,7 @@ def parse_git_binary_diff(text):
                 continue
         elif old_size > 0:
             if base85string.match(line):
-                assert len(line) >= 7 and ((len(line) - 2) % 5) != 0
+                assert len(line) >= 6 and ((len(line) - 2) % 5) != 0
```
Comment (babenek) [inline src/whatthepatch/patch.py:995]:
```suggestion
                assert len(line) >= 6 and ((len(line) - 1) % 5) == 0
```

**Review by abbradar — APPROVED** [review]
Kudos for the test! And thanks for the fix :)

File: tests/test_patch.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-
+import hashlib
 import os
 import time
 import unittest
```
Comment (cscorley) [inline tests/test_patch.py:5]: In the old test, something weird is up with line 1453, I get a DeprecationWarning on it:

```
========================================== test session starts ===========================================
platform linux -- Python 3.10.9, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/cscorley/git/whatthepatch
collected 52 items

tests/test_apply.py .............                                                                  [ 25%]
tests/test_patch.py .......................................                                        [100%]

============================================ warnings summary ============================================
tests/test_patch.py:1453
  /home/cscorley/git/whatthepatch/tests/test_patch.py:1453: DeprecationWarning: invalid escape sequence '\ '
    text = """---

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================== 52 passed, 1 warning in 3.09s ======================================
```

It happens on main, too, so it's fine; just FYI for next time you're around this :)

**Comment by cscorley** [comment]
Thanks to the both of you!

**Review by cscorley — APPROVED** [review]

File: tests/test_patch.py (hunk 3 above)
Comment (cscorley) [reply tests/test_patch.py:5]: Interesting -- I can no longer trigger this after merging.

Comment (cscorley) [reply tests/test_patch.py:5]: Curiosity got the best of me, I figured it out.  Just need to put these into separate files as in PR #58

**Pull request merged**